mango-core/src/main/scala/org/bdgenomics/mango/models/TrackedLayout.scala

```diff
@@ -139,15 +191,47 @@ class OrderedTrackedLayout[T: ClassTag](values: Traversable[(ReferenceRegion, T)
   class Track(val initial: (ReferenceRegion, T)) {
 
     val records = new mutable.ListBuffer[(ReferenceRegion, T)]()
-    records += initial
+    val groupPairs = new mutable.ListBuffer[GroupPair]()
+    val idx: Int = numTracks
+
+    if (initial != null)
+      records += initial
+
+    def this() {
+      this(null)
+    }
 
     def +=(rec: (ReferenceRegion, T)): Track = {
       records += rec
       this
     }
 
+    def addGroupPair(recs: Seq[(ReferenceRegion, T)]) {
+      if (recs.size < 2)
```

Comment:
I believe unpaired reads aren't being dropped. All reads are added to a track, and then are checked to see if there exist group pairs. Unpaired reads are still stored in records (line 193), and any reads that are paired are additionally stored in groupPairs (line 194).